src/field_theory/minimal_polynomial.lean

```diff
@@ -65,6 +65,132 @@ ne_zero_of_monic (monic hx)
 
 end ring
 
+section integral_domain
+
+variables [integral_domain α]
+
+section ring
+
+variables [ring β] [algebra α β] [nontrivial β]
+variables {x : β} (hx : is_integral α x)
+
+/--The degree of a minimal polynomial is nonzero.-/
+lemma degree_ne_zero : degree (minimal_polynomial hx) ≠ 0 :=
+begin
+  assume deg_eq_zero,
+  have ndeg_eq_zero : nat_degree (minimal_polynomial hx) = 0,
+  { simpa using congr_arg nat_degree (eq_C_of_degree_eq_zero deg_eq_zero) },
+  have eq_one : minimal_polynomial hx = 1,
+  { rw eq_C_of_degree_eq_zero deg_eq_zero, convert C_1,
+    simpa [ndeg_eq_zero.symm] using (monic hx).leading_coeff },
+  simpa [eq_one, aeval_def] using aeval hx
+end
+
+/--A minimal polynomial is not a unit.-/
+lemma not_is_unit : ¬ is_unit (minimal_polynomial hx) :=
+assume H, degree_ne_zero hx $ degree_eq_zero_of_is_unit H
+
+end ring
+
+section domain
+
+variables [domain β] [algebra α β]
+variables {x : β} (hx : is_integral α x)
+
+/--A minimal polynomial is irreducible.-/
+lemma irreducible : irreducible (minimal_polynomial hx) :=
+begin
+  cases irreducible_or_factor (minimal_polynomial hx) (not_is_unit hx) with hirr hred,
+  { exact hirr },
+  obtain ⟨a, hred⟩ := hred,
+  obtain ⟨b, hred⟩ := hred,
```

Comment:
Or possibly (untested):
```suggestion
  obtain (rfl, ⟨a, b, hred⟩) := irreducible_or_factor (minimal_polynomial hx) (not_is_unit hx),
```

Reply:
This does not seem to work.

Reply:
Maybe try it with `hirr` instead of `rfl`?

Reply:
I can replace
```
cases irreducible_or_factor (minimal_polynomial hx) (not_is_unit hx) with hirr hred,
{ exact hirr },
exfalso,
--  obtain ⟨a, b, ha_nunit, hb_nunit, hab_eq⟩ := hred,
```
with
```
obtain (hirr | ⟨a, b, ha_nunit, hb_nunit, hab_eq⟩) := irreducible_or_factor (minimal_polynomial hx)
(not_is_unit hx),
{ exact hirr },
exfalso,
```
But I don't know if it is really better.

Reply:
If it means you need to wrap a line, then I wouldn't bother, and keep it as it is.